pallets/staking/src/pallet/mod.rs

```diff
@@ -0,0 +1,2113 @@
+    /// range [0, Cap].
+    #[pallet::storage]
+    #[pallet::getter(fn validator_commission_cap)]
+    pub type ValidatorCommissionCap<T: Config> = StorageValue<_, Perbill, ValueQuery>;
+
+    /// The minimum amount with which a validator can bond.
+    #[pallet::storage]
+    #[pallet::getter(fn min_bond_threshold)]
+    pub type MinimumBondThreshold<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+    // Slashing switch for validators & Nominators.
+    #[pallet::storage]
+    #[pallet::getter(fn slashing_allowed_for)]
+    pub type SlashingAllowedFor<T: Config> = StorageValue<_, SlashingSwitch, ValueQuery>;
+
+    // -------------------------------------------------------------
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        pub validator_count: u32,
+        pub minimum_validator_count: u32,
+        pub invulnerables: Vec<T::AccountId>,
+        pub force_era: Forcing,
+        pub slash_reward_fraction: Perbill,
+        pub canceled_payout: BalanceOf<T>,
+        pub stakers: Vec<(
+            IdentityId,
+            T::AccountId,
+            T::AccountId,
+            BalanceOf<T>,
+            crate::StakerStatus<T::AccountId>,
+        )>,
+        pub validator_commission_cap: Perbill,
+        pub min_bond_threshold: BalanceOf<T>,
+        pub slashing_allowed_for: SlashingSwitch,
+        pub history_depth: u32,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            GenesisConfig {
+                validator_count: Default::default(),
+                minimum_validator_count: Default::default(),
+                invulnerables: Default::default(),
+                force_era: Default::default(),
+                slash_reward_fraction: Default::default(),
+                canceled_payout: Default::default(),
+                stakers: Default::default(),
+                validator_commission_cap: Default::default(),
+                min_bond_threshold: Default::default(),
+                slashing_allowed_for: Default::default(),
+                history_depth: Default::default(),
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+        fn build(&self) {
+            ValidatorCount::<T>::put(self.validator_count);
+            MinimumValidatorCount::<T>::put(self.minimum_validator_count);
+            Invulnerables::<T>::put(&self.invulnerables);
+            ForceEra::<T>::put(self.force_era);
+            SlashRewardFraction::<T>::put(self.slash_reward_fraction);
+            CanceledSlashPayout::<T>::put(self.canceled_payout);
+            ValidatorCommissionCap::<T>::put(self.validator_commission_cap);
+            MinimumBondThreshold::<T>::put(self.min_bond_threshold);
+            SlashingAllowedFor::<T>::put(self.slashing_allowed_for);
+            //HistoryDepth::<T>::put(self.history_depth);
```

Comment:
Shouldn't we read this from config and add this write?

Reply:
All the genesis config logic will not have any impact on running networks (Staging, Testnet, Mainnet).  Only local dev chains starting a new fresh chain.  So I think it is safe to enable this code.

Reply:
Looks like we don't really need to initialize `HistoryDepth`, since it is using `HistoryDepthDefault` as it's default value.

Either:
1. Use the `history_depth` value from  the genesis config and remove `fn HistoryDepthDefault`.
2. Or remove `history_depth` from the genesis config, since `HistoryDepthDefault` will provide a value.

To keep things closer to what our current code does, I recommend  #1.  The next PR can move `HistoryDepth` to a constant, like the newer staking code.